packages/clients/diplan/src/store/utils/mergeMultiGeometries.ts

```diff
@@ -0,0 +1,71 @@
+import { Feature, FeatureCollection, GeoJsonTypes } from 'geojson'
+
+type GeometryType = Exclude<
+  GeoJsonTypes,
+  'Feature' | 'FeatureCollection' | 'GeometryCollection'
+>
+
+const isMulti = (type: string) => type.startsWith('Multi')
+const multi = (type: string): string => (isMulti(type) ? type : `Multi${type}`)
+
+const mergeBin = (features: Feature[]): [Feature] | [] =>
+  features.length === 0
+    ? []
+    : [
+        {
+          ...features[0],
+          geometry: {
+            type: multi(features[0].geometry.type) as GeometryType,
+            coordinates: [
+              ...features.map(({ geometry }) =>
+                isMulti(geometry.type)
+                  ? // @ts-expect-error | We know it's no GeometryCollection
+                    geometry.coordinates[0]
+                  : // @ts-expect-error | We know it's no GeometryCollection
+                    geometry.coordinates
+              ),
+            ],
+          },
+        },
+      ]
```

Comment:
```suggestion
import {
  Feature,
  FeatureCollection,
  GeoJsonTypes,
  Geometry,
  GeometryCollection,
} from 'geojson'

type Type = Exclude<Geometry, GeometryCollection>

const isMulti = (type: string) => type.startsWith('Multi')
const multi = (type: string): string => (isMulti(type) ? type : `Multi${type}`)

const mergeBin = (features: Feature<Type>[]): [Feature<Type>] | [] =>
  features.length === 0
    ? []
    : [
        {
          ...features[0],
          geometry: {
            type: multi(features[0].geometry.type),
            coordinates: [
              ...features.map(({ geometry }) =>
                isMulti(geometry.type)
                  ? geometry.coordinates[0]
                  : geometry.coordinates
              ),
            ],
          },
        } as Feature<Type>,
      ]
```
This fixes the issue. `GeoJsonTypes` included strings whereas `Feature` expects a `Geometry`

Reply:
Yes, but in a more real sense, no. `as Feature<Type>` is just another flavour of `// @ts-expect-error` in my book. However, with the input around it, I managed to proceed further and narrowed the issue down to this: https://github.com/Dataport/polar/pull/245/commits/a75aeef3b2869532c2ba977da5196f783995bd73

```
result[0].geometry.coordinates = [
    features.map(({ geometry }) => {
      if (isMulti(geometry.type)) {
        return geometry.coordinates[0]
      }
      return geometry.coordinates
    }) as Position, // actually also (... | Position[] | Position[][]), but can't | those types here for reasons unknown, albeit every single one works
```

And now I'm really confused. `Position`? Works. `Position[]`? Works. `Position[][]`? Works. `Position | Position[] | Position[][]`? Well, that's a paddlin'. I get why TS can't infer the type dependant on `isMulti` here properly since it'd have to somewhat execute the ternary statement and probably stops at that point to not melt my hardware, so I do understand why casting is not optional here.

I *think* I'm now at the root of the issue, but can't resolve it without the lie that `as Position`  is. (Or, at least, it's just a part of the truth ...)